core/templates/components/state-editor/state-skill-editor/state-skill-editor.directive.ts

```diff
@@ -0,0 +1,135 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
```

Comment:
Is it OK to write this in AngularJS rather than Angular? /cc @srijanreddy98

Reply:
Hey @srijanreddy98, PTAL!

Reply:
No. It's not... unless there is a service that's used here and is in AJS. Please use angular to make components. Another thing I would like to point out is that when creating files, make sure they are well-typed. There are a lot of implied any in this code. Such code can be an impediment to migration.

Reply:
Let me know if you need any help with / have questions about writing this component in angular. Happy to help.

Reply:
@srijanreddy98, this file uses a controller -- SelectSkillModalController which is in AJS. So should I migrate that to Angular too?